packages/scheduler/src/SchedulerProfiling.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {PriorityLevel} from './SchedulerPriorities';
+import {enableProfiling} from './SchedulerFeatureFlags';
+
+import {NoPriority} from './SchedulerPriorities';
+
+let runIdCounter: number = 0;
+let mainThreadIdCounter: number = 0;
+
+const profilingStateSize = 4;
+export const sharedProfilingBuffer = enableProfiling
+  ? // $FlowFixMe Flow doesn't know about SharedArrayBuffer
+    typeof SharedArrayBuffer === 'function'
+    ? new SharedArrayBuffer(profilingStateSize * Int32Array.BYTES_PER_ELEMENT)
+    : // $FlowFixMe Flow doesn't know about ArrayBuffer
+      typeof ArrayBuffer === 'function'
+      ? new ArrayBuffer(profilingStateSize * Int32Array.BYTES_PER_ELEMENT)
+      : null // Don't crash the init path on IE9
+  : null;
+
+const profilingState =
+  enableProfiling && sharedProfilingBuffer !== null
+    ? new Int32Array(sharedProfilingBuffer)
+    : null;
```

Comment:
Why not just fallback to a regular array?

Reply:
shrug.png

Reply:
That's an interesting approach because it would allow us to remove the null checks below and it kind of keeps working, except it wouldn't be able to be introspected since there's no buffer exposed. I like it just for byte savings though.

Reply:
Is there any harm to always doing `new Int32Array(sharedProfilingBuffer)`? Even when it's off? We'd just not write to it when profiling is off. And when buffer is not supported, we'd pass `null` which would give us an empty array.

I guess it doesn't work in IE9 though so probably should be guarded.